classes/plugins/DOIPubIdExportPlugin.inc.php

```diff
@@ -165,6 +340,61 @@ function getPubIdDisplayType() {
 	}
 
 	/**
+	 * Get the XML for selected objects.
+	 * @param $objects array Array of published articles, issues or galleys
+	 * @param $filter string
+	 * @param $context Context
+	 * @return string XML contents representing the supplied DOIs.
+	 */
+	function exportXML($objects, $filter, $context) {
+		$xml = '';
+		$filterDao = DAORegistry::getDAO('FilterDAO');
+		$exportFilters = $filterDao->getObjectsByGroup($filter);
+		assert(count($exportFilters) == 1); // Assert only a single serialization filter
+		$exportFilter = array_shift($exportFilters);
+		$exportDeployment = $this->_instantiateExportDeployment($context);
+		$exportFilter->setDeployment($exportDeployment);
+		libxml_use_internal_errors(true);
+		$exportXml = $exportFilter->execute($objects, true);
+		$xml = $exportXml->saveXml();
+		$errors = array_filter(libxml_get_errors(), create_function('$a', 'return $a->level == LIBXML_ERR_ERROR ||  $a->level == LIBXML_ERR_FATAL;'));
+		if (!empty($errors)) {
+			$this->displayXMLValidationErrors($errors, $xml);
+			fatalError(__('plugins.importexport.common.error.validation'));
+		}
+		return $xml;
+	}
+
+	/**
+	 * Deposit XML i.e. publication objects DOIs.
+	 * This must be implemented in the subclasses, if the action is supported.
+	 * @param $objects array Array of published articles, issues or galleys
+	 * @param $context Context
+	 * @param $filename Export XML filename
+	 * @return boolean Weather the DOI has been registered/found
+	 */
+	abstract function depositXML($objects, $context, $filename);
+
+	/**
+	 * Check statuses for selected publication objects.
+	 * @param $objects array Array of published articles, issues or galleys
+	 * @param $context Context
+	 */
+	function checkStatus($objects, $context) {
+		foreach ($objects as $object) {
+			$this->updateDepositStatus($context, $object);
+		}
+	}
+
+	/**
+	 * Update deposit status.
+	 * The function must be implemented in the subclasses, if the action is supported.
+	 * @param $context Context
+	 * @param $object The object getting deposited
+	 */
+	function updateDepositStatus($context, $object) { }
```

Comment:
Hmmm... this is implemented only in Crossref plugin, but used in checkStatus function in this class. CheckStatus is also used only in Crossref plugin, thus I think it is best to remove it from here and put it in the CrossrefExportPlugin class.